Microsoft.SqlVirtualMachine new API 2023-10-01 Stable

yareyes opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![diagram](https://github.com/Azure/azure-rest-api-specs/assets/20296335/35140fb9-639e-4e44-ac4b-38b1a20dddf9)

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- For help with Step 2 / "ARM review", see https://aka.ms/azsdk/pr-arm-review.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

yareyes committed:
Adds base for updating Microsoft.SqlVirtualMachine from version preview/2023-01-01-preview to version 2023-10-01

yareyes committed:
Updates readme

yareyes committed:
Updates API version in new specs and examples

yareyes committed:
Adding new 2023-10-01 API

yareyes committed:
Adding fixes for errors

yareyes commented:
/azp run

yareyes committed:
Updating a spell check.

yareyes committed:
Adding schema version for 200 response

yareyes committed:
Removed the specification

yareyes committed:
Added OsType

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes committed:
Updated to PATCH parameters

yareyes committed:
Removing hidden properties

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes committed:
Updated Swagger to latest

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes committed:
Update 200 response codes

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes committed:
Removing 200 response

yareyes committed:
Removing 200 response from example files

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

rkmanda commented:
>           "202": {

The final response of the POST action should be documented with a 200 status code.

---
Refers to: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/stable/2023-10-01/sqlvm.json:776 in 11ced6c. [](commit_id = 11ced6c6ef03578455fda6a5a012ed402160ad5c, deletion_comment = False)

rkmanda commented:
>           "description": "Resource Id of the identity. Only required when userAssigned identity is selected.",

apply format: arm-id

---
Refers to: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/stable/2023-10-01/sqlvm.json:2363 in 11ced6c. [](commit_id = 11ced6c6ef03578455fda6a5a012ed402160ad5c, deletion_comment = False)

rkmanda reviewed (changes requested):
:clock1:

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes committed:
Update description on the virtual machine resource id.

yareyes committed:
Merge branch 'yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01' of https://github.com/Azure/azure-rest-api-specs into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes commented:
@rkmanda 
For the change: arm-id. I have changed the comments that now say "ARM Resource Id of the identity". 

Now for the POST comment "The final response of the POST action should be documented with a 200 status code so the client can deserialize the final response." 

So all our POSTs (or 99% of them) - return a 200, but it's an empty object. When adding the response 200: { }, swagger validation complains that empty objects should be removed from the response, so we removed all of them, else validations will start failing. We wanted to keep the same consistency in all the POSTs, as you see all other POST api's have the same behavior. Can we continue this behavior or do we have to change the behavior for everything?

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

gary-x-li commented:
>           "description": "Resource Id of the identity. Only required when userAssigned identity is selected.",

What Roopesh meant was adding the following property: `"format": "arm-id"`. [https://azure.github.io/autorest/extensions/#x-ms-arm-id-details](https://azure.github.io/autorest/extensions/#x-ms-arm-id-details)

---
In reply to: [2038264429](https://github.com/Azure/azure-rest-api-specs/pull/27782#issuecomment-2038264429) [](http://example.com/codeflow?ancestors=2038264429)

---
Refers to: specification/sqlvirtualmachine/resource-manager/Microsoft.SqlVirtualMachine/stable/2023-10-01/sqlvm.json:2363 in 11ced6c. [](commit_id = 11ced6c6ef03578455fda6a5a012ed402160ad5c, deletion_comment = False)

yareyes committed:
Update the arm-id

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

Alancere committed:
add sdk-suppressions.yaml

Alancere committed:
fix suppression

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01

yareyes commented:
@rkmanda since ARM has signed off, could you approve this PR or remove yourself from reviewers? Not sure what is the right process here.

yareyes commented:
/pr RequestMerge

raosuhas commented:
You need to fix the merge conflict to checkin

yareyes committed:
Merge branch 'main' into yareyes-sqlvirtualmachine-Microsoft.SqlVirtualMachine-2023-10-01
